sdk/database/helper/connutil/sql.go

```diff
@@ -107,6 +135,18 @@ func (c *SQLConnectionProducer) Init(ctx context.Context, conf map[string]interf
 		return nil, errwrap.Wrapf("invalid max_connection_lifetime: {{err}}", err)
 	}
 
+	if c.AuthType == AuthTypeGCPIAM {
```

Comment:
Should we do anything to prevent other DB types that don't support CloudSQL from setting this? e.g. cassandra, influx, etc. Maybe we are and I missed it?

Reply:
it would technically fail at `registerDrivers()` but we can elevate it to sooner.

Reply:
I am thinking we should not fail if `auth_type` is provided. Typically, Vault will ignore unsupported fields and carry on. I am happy to discuss this in more detail if you like.

Reply:
We could do that too - just ignore the field if the combination is nonsensical "mongodb" + "gcp_iam" is just mongodb

Do we want to emit a warning, or just keep going silently with what we 'know' you meant?

Reply:
I thought about this a bit more. I think we should do some validation when `auth_type` and `credentials` are provided for DBs that don't support those fields. However, this should not block this PR and we can punt it to a later date. What do you think?

Reply:
We can punt, I think, since I don't think we're currently in a state where an inappropriate configuration could end up committed.

Reply:
We decided to punt on this to a later PR in the future. For now, on an unrecognized combination of `auth_type` and DB type, we will error with a meaningful message to the console. Resolving this conversation for now 🙏🏼